### PR TITLE
Reset the "jobs" config variable when upgrading to opam 2.2

### PR DESCRIPTION
**Pull request by kit-ty-kate**
Fixes https://github.com/ocaml/opam/issues/5816

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
Note from the dev meeting: a better fix would be to upgrade the root-version to `2.2~beta` and move the upgrade there instead to prevent issues for (rare) people who would have upgraded their opam all the way from 2.0 to 2.1 to 2.2.0~alpha (in which case this code would not have been run)

File: src/state/opamFormatUpgrade.ml [hunk 1]
```diff
@@ -1130,12 +1134,7 @@ let from_2_0_to_2_1 ~on_the_fly _ conf =
    | Some prev_jobs when prev_jobs = max 1 (OpamSysPoll.cores () - 1) -> ()
    | Some prev_jobs -> info_jobs_changed ~prev_jobs
    | None -> info_jobs_changed ~prev_jobs:1);
-  let conf = apply_eval_variables conf [] sys_config_variables_unix_2_1 in
```
Comment (kit-ty-kate) [inline src/state/opamFormatUpgrade.ml:1133]: issue: this transformation is lost

**Review by rjbou — COMMENTED** [review]
Thanks! Agree with your comment, we'll see when we upgrade the root number.

File: src/state/opamFormatUpgrade.ml (hunk 1 above)
Comment (kit-ty-kate) [reply src/state/opamFormatUpgrade.ml:1133]: fixed

**Comment by kit-ty-kate** [comment]
Ready to be reviewed again

File: tests/reftests/opamroot-versions.test [hunk 2]
```diff
@@ -2712,8 +2746,8 @@ compiler: ["i-am-sys-compiler.2"]
 installed: ["i-am-package.2" "i-am-sys-compiler.2"]
 opam-version: "2.0"
 roots: ["i-am-package.2" "i-am-sys-compiler.2"]
-### :V:6:e: reinit from 2.2~alpha
-### ocaml generate.ml 2.2~alpha
+### :V:6:e: reinit from 2.2~beta
```
Comment (rjbou) [inline tests/reftests/opamroot-versions.test]: At each bump root, a new entry should be added, I updated the PR

**Review by rjbou — COMMENTED** [review]
Once the reftest reviewed, good to go!
thanks!

File: tests/reftests/opamroot-versions.test (hunk 2 above)
Comment (kit-ty-kate) [reply tests/reftests/opamroot-versions.test]: Thanks! That looks good to me

**Review by rjbou — APPROVED** [review]
As the tests looks good, unblocking :)

**Pull request merged**